### DISCUSSION TITLE
Asynchronous write from a list

### DISCUSSION THREAD

**Discussion by Hussain-f** [osiegmar/FastCSV#114] [Q&A] [answered]
Does FastCSV support asynchronous writing. I have a list of records that that I want to parallel stream and asynchronously write each record from the stream to a CSV file.
From the docs, it seems async reads are supported but not writes...

**Comment by osiegmar** [comment, marked as answer]
If using FastCSV in a multithreaded environment, FastCSV requires external synchronization to ensure correct behavior. 

> From the docs, it seems async reads are supported but not writes...

What docs are you exactly referring to?
